src/amplitude-client.js

```diff
@@ -166,12 +166,12 @@ var _parseConfig = function _parseConfig(options, config) {
 
   // validates config value is defined, is the correct type, and some additional value sanity checks
   var parseValidateAndLoad = function parseValidateAndLoad(key) {
-    if (!DEFAULT_OPTIONS.hasOwnProperty(key)) {
+    if (!options.hasOwnProperty(key)) {
```

Comment:
removing references to `DEFAULT_OPTIONS` to allow recursive calling of `_parseConfig`